### PR TITLE
release: 1.3.5 build 80

### PR DESCRIPTION
**Pull request by dcposch**
**Faster, smoother, invite codes, bridge onramp.**

### API and webapp

- [x] Branch from `master`
- [x] Testnet API deployed correctly **now with proper healthcheck, waits for indexer sync**
- [x] ~Testnet webapp deployed correctly~ none yet
- [x] Logs look clean
- [x] Test Honeycomb looks clean

### Backcompat

- [x] Push to `prod`
- [x] Prod API deploys correctly
- [x] Prod website deploys correctly
- [x] Logs clean
- [x] Honeycomb clean
- [x] Ensure old deeplinks still work.
- [x] Using *old* app (no invCode), create new account. Should not send faucet.
- [x] Test bridge onramp.
- [x] Using *old* app, use existing account.

### iOS

Watch all of the following interactions closely for jank and UX regressions, not
just outright bugs.

- [x] Install from private TestFlight
- [x] Create account
- [x] Request faucet
- [x] Notification appears on faucet send
- [x] Send payment
- [x] Create Payment Link, open in browser
- [x] Cancel Payment Link
- [x] Tap Payment Link, ensure it opens in-app
- [x] Create Request, open in browser
- [x] Add device + Use existing onboarding (test transactions work on both devices)
- [x] Remove device
- [x] Open History
- [x] Tap transaction, view in block explorer
- [x] Account screen: Send Debug Log
- [x] Debug log looks clean

### Android

- [x] Install from Play Store closed track
- [x] Clear account (create one first, if necessary)
- [x] Create account
- [x] Request faucet
- [x] Notification appears on faucet send
- [x] Send payment
- [x] Create Payment Link, open in browser
- [x] Cancel Payment Link
- [x] Tap Payment Link, ensure it opens in-app
- [x] Add device + Use existing onboarding (test transactions work on both devices)
- [x] Remove device
- [x] Create Request, open in browser
- [x] Open History
- [x] Tap transaction, view in block explorer
- [x] Account screen: Send Debug Log
- [x] Debug log looks clean

### Promote release

BEFORE merging this PR,

- [x] Push to public TestFlight + Play Store open test track + App Store + Play Store

Production mainnet releases will eventually trail TestFlight by at least a week
to allow longer and more thorough testing.

### CODE REVIEW

File: apps/daimo-mobile/src/view/screen/send/SendTransferButton.tsx [hunk 1]
```diff
@@ -45,20 +45,25 @@ function SendTransferButtonInner({
   requestId,
 }: SendTransferButtonProps & { account: Account }) {
   console.log(`[SEND] rendering SendButton ${dollars} ${requestId}`);
+
+  // Get exact amount. No partial cents.
   assert(dollars >= 0);
+  const maxDecimals = 2;
+  const dollarsStr = dollars.toFixed(maxDecimals) as `${number}`;
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/send/SendTransferButton.tsx:52]: not strictly necessary, but here as a second layer of validation

File: apps/daimo-mobile/src/view/shared/AmountInput.tsx [hunk 2]
```diff
@@ -74,16 +77,26 @@ function AmountInput({
   const change = useCallback((text: string) => {
     if (disabled) return;
 
-    setStrVal(text);
-    const newVal = parseLocalFloat(text);
+    // Validate. Handle negative numbers, NaN, out of range.
+    const looksValid = /^(|0|(0[.,]\d*)|([1-9]\d*[.,]?\d*))$/.test(text);
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/AmountInput.tsx:81]: took a while to get this right

tested with both the US `0.00` and UK `0,00` locale

File: apps/daimo-mobile/src/view/shared/AmountInput.tsx [hunk 3]
```diff
@@ -9,12 +9,15 @@ import {
...
 } from "react-native";
 
-import { getAmountText } from "./Amount";
+import { amountSeparator, getAmountText } from "./Amount";
 import Spacer from "./Spacer";
 import { color, ss } from "./style";
 import { TextCenter, TextLight } from "./text";
 import { useAccount } from "../../model/account";
 
+// Input componens allows entry in range $0.01 to $99,999.99
```
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/view/shared/AmountInput.tsx:18]: nit `components`

Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/shared/AmountInput.tsx:18]:
```suggestion
// Input components allows entry in range $0.01 to $99,999.99
```

**Review by nalinbhardwaj — APPROVED** [review]

File: apps/daimo-mobile/src/view/shared/AmountInput.tsx (hunk 3 above)
Comment (dcposch) [reply apps/daimo-mobile/src/view/shared/AmountInput.tsx]: shouldn've said `component` all good

**Pull request merged**